src/Filesystem/Folder.php

```diff
@@ -194,16 +218,27 @@ public function read($sort = true, $exceptions = false, $fullPath = false)
             if ($fullPath) {
                 $name = $item->getPathname();
             }
+
             if ($item->isDir()) {
-                $dirs[] = $name;
+                $dirs[$item->{$methodName}()][] = $name;
             } else {
-                $files[] = $name;
+                $files[$item->{$methodName}()][] = $name;
             }
         }
+
         if ($sort || $this->sort) {
-            sort($dirs);
-            sort($files);
+            ksort($dirs);
+            ksort($files);
         }
+
+        if (!empty($dirs)) {
```

Comment:
Should be either `!$dirs` or `count(...) === 0`.
!empty() cloaks spelling mistakes.